pkg/tasker/tasker.go

```diff
@@ -0,0 +1,196 @@
+	tellorFilterer, err := tellor.NewITellorFilterer(mt.contractInstance.Address, mt.client)
+	if err != nil {
+		return nil, nil, errors.Wrap(err, "getting ITellorFilterer instance")
+	}
+	sub, err := tellorFilterer.WatchNewChallenge(&bind.WatchOpts{}, sink, nil)
+	if err != nil {
+		return nil, nil, errors.Wrap(err, "getting NewChallenge channel")
+	}
+	return sink, sub, nil
+}
+
+func (mt *MiningTasker) subscribeToNewChallenge() error {
+	sink, sub, err := mt.getNewChallengeChannel()
+	if err != nil {
+		return err
+	}
+	level.Info(mt.logger).Log("msg", "subscribed to NewChallenge events")
+
+	go func() {
+		for {
+			select {
+			case <-mt.done:
+				sub.Unsubscribe()
+				level.Info(mt.logger).Log("msg", "unsubscribed to NewChallenge events")
+				return
+			case err := <-sub.Err():
+				if err != nil {
+					level.Error(mt.logger).Log(
+						"msg",
+						"new challenge subscription error",
+						"err", err)
+				}
+				mt.resubscribe <- true
+			case vLog := <-sink:
+				mt.sendWork(vLog)
+			}
+		}
+	}()
+	return nil
+}
+
+func (mt *MiningTasker) sendWork(challenge *tellor.ITellorNewChallenge) {
+	if challenge.CurrentRequestId[0].Int64() > int64(100) || challenge.CurrentRequestId[0].Int64() == 0 {
+		level.Warn(mt.logger).Log("msg", "new current variables request ID not correct - contract about to be upgraded")
+		return
+	}
+	work := mt.CreateWork(challenge)
+	// Send new work to the sink.
+	if work != nil {
+		mt.workSink <- work
+	}
+}
+
+func (mt *MiningTasker) Start() error {
+	level.Info(mt.logger).Log("msg", "tasker has been started")
+	currentChallenge, err := mt.getCurrentChallenge()
+	if err != nil {
+		return errors.Wrap(err, "tasker getting the current challenge")
+	}
+	level.Info(mt.logger).Log("msg", "tasker is sending the initial challenge to the miner")
+	mt.sendWork(currentChallenge)
+	err = mt.subscribeToNewChallenge()
+	if err != nil {
+		return errors.Wrap(err, "tasker subscribing to new challenges")
+	}
+	for {
+		select {
+		case <-mt.resubscribe:
+			err = mt.subscribeToNewChallenge()
+			if err != nil {
+				return errors.Wrap(err, "tasker resubscribing to new challenges")
+			}
+		case <-mt.ctx.Done():
+			mt.done <- true
+		}
+	}
+}
+
+func (mt *MiningTasker) Stop() {
+	mt.close()
+	level.Info(mt.logger).Log("msg", "tasker shutdown complete")
+}
+
+func (mt *MiningTasker) CreateWork(challenge *tellor.ITellorNewChallenge) *mining.Work {
+	newChallenge := &mining.MiningChallenge{
+		Challenge:  challenge.CurrentChallenge[:],
+		Difficulty: challenge.Difficulty,
+		RequestIDs: challenge.CurrentRequestId,
+	}
+
+	level.Debug(mt.logger).Log("msg", "new challenge for mining",
+		"hex", fmt.Sprintf("%x", newChallenge.Challenge),
+		"difficulty", newChallenge.Difficulty,
+		"requestIDs", fmt.Sprintf("%+v", newChallenge.RequestIDs),
+	)
+
+	mt.currChallenge = newChallenge
+	return &mining.Work{Challenge: newChallenge, PublicAddr: mt.accounts[0].Address.String(), Start: uint64(rand.Int63()), N: math.MaxInt64}
+}
```

Comment:
Don't need this function anymore and can go directly in `sendWork`
Also can remove `mt.currChallenge = newChallenge` this is from the old workflow.
```suggestion

}
```